### PR TITLE
Fixes #85 - :deploy goal should quietly skip non-war projects

### LINKED ISSUE #85: :deploy goal should quietly skip non-war projects
If you attempt to setup a common `appengine:deploy` from a parent pom, the maven plugin fails as its trying to deploy the `<packaging>pom</packaging>` project.

If it encounters a non-war packaging, it should just skip the `appengine:deploy` attempt. 

[ERROR] Failed to execute goal com.google.cloud.tools:appengine-maven-plugin:0.1.2:deploy (deploy-integration-test) on project tests-webapps-parent: Execution deploy-integration-test of goal com.google.cloud.tools:appengine-maven-plugin:0.1.2:deploy failed: Artifact doesn't exist at '/home/joakim/code/jetty/gcloud.project/jetty-runtime/tests/webapps/target/tests-webapps-parent-0.1.0-SNAPSHOT.pom' -> [Help 1]

### PR DESCRIPTION
**Pull request by akerekes**

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java [hunk 1]
```diff
@@ -43,6 +43,9 @@
   @Parameter(defaultValue = "${project}", readonly = true)
   protected MavenProject mavenProject;
 
+  @Parameter( property = "packaging" )
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java]: Spacing?

File: src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java [hunk 2]
```diff
@@ -92,4 +95,8 @@ public String getCompileTargetVersion() {
     }
     return javaVersion;
   }
+
+  protected Plugin getPlugin() {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java]: I wonder if there's a way to get there from maven?

File: src/main/java/com/google/cloud/tools/maven/DeployMojo.java [hunk 3]
```diff
@@ -90,6 +90,10 @@
 
   @Override
   public void execute() throws MojoExecutionException, MojoFailureException {
+    if (!"war".equals(packaging) && !"jar".equals(packaging)) {
+      // https://github.com/GoogleCloudPlatform/app-maven-plugin/issues/85
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/DeployMojo.java]: Should we print a message?

File: src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java (hunk 1 above)
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java]: Also is this just filled in by maven?

Comment (akerekes) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java]: Spacing fixed. And yes, this is filled in by Maven.

File: src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java (hunk 2 above)
Comment (akerekes) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java]: Actually we can get it from the `pluginDescriptor` field. In any case I've removed it as it's not used anywhere, probably leftover from some fiddling around.

**Pull request merged**